src/components/02_atoms/MultipleFields/MultipleFields.js

```diff
@@ -46,14 +49,6 @@ class MultipleFields extends Component {
     newItemAdded: false,
   };
 
-  componentDidMount() {
```

Comment:
So yeah, instead of calling to onChange() we construct the value just for rendering purposes, see below

Reply:
The onChange() call caused the store to override the previous restorable value.